add aws sqs transport (#492)
openzipkin/zipkin-js: packages/zipkin-transport-aws-sqs/README.md

# Zipkin-transport-aws-sqs

![npm](https://img.shields.io/npm/dm/zipkin-transport-aws-sqs.svg)

This is a module that sends Zipkin trace data from zipkin-js to AWS SQS.

## Usage

`npm install zipkin-transport-aws-sqs --save`

```javascript
const {
  Tracer,
  BatchRecorder,
  jsonEncoder: {JSON_V2}
} = require('zipkin');
const {AwsSqsLogger} = require('zipkin-transport-aws-sqs');
const noop = require('noop-logger');

const AwsSqsRecorder = new BatchRecorder({
  logger: new AwsSqsLogger({
    queueUrl: "https://...", //mandatory
    awsConfig: {
      credentials: {},
      accessKeyId: '...',
      region: '...',
      secretAccessKey: '...',
      credentialProvider: {}
    }, //optional
    delaySeconds: 0,// optional

>     encoder: JSON_V2, // optional (defaults to JSON_V2)

Reviewer: I'd remove this parameter and make this to work with JSON_V2 until someone asks for the feature.

# Zipkin-transport-aws-sqs

![npm](https://img.shields.io/npm/dm/zipkin-transport-aws-sqs.svg)

This is a module that sends Zipkin trace data to Amazon SQS for collection and processing.

> SQS is an alternative to kafka that is fully managed in the AWS cloud.

Reviewer 2: ```suggestion
SQS is an alternative to Apache Kafka that is fully managed in the AWS cloud.
```

## Usage

`npm install zipkin-transport-aws-sqs --save`

```javascript
const {
  Tracer,
  BatchRecorder
} = require('zipkin');
const {AwsSqsLogger} = require('zipkin-transport-aws-sqs');
const noop = require('noop-logger');

let awsSqsLogger = new AwsSqsLogger({
                         queueUrl: "https://...", //mandatory

>                          endpointConfiguration: AWS.Endpoint,// optional

Reviewer 2: ```suggestion
                         endpointConfiguration: AWS.Endpoint, // optional
```

>                          region: 'eu-west-1', // optional, region string

Reviewer: If this is optional, what is the default value?

>                          credentialProvider: AWS.CredentialProviderChain,// optional

Reviewer 2: ```suggestion
                         credentialProvider: AWS.CredentialProviderChain, // optional
```

>                          delaySeconds: 0,// optional

Reviewer 2: ```suggestion
                         delaySeconds: 0, // optional
```

                         log: noop // optional (defaults to console)
                       });

const AwsSqsRecorder = new BatchRecorder({
  logger: awsSqsLogger
});

const tracer = new Tracer({
  recorder,
  ctxImpl, // this would typically be a CLSContext or ExplicitContext
  localServiceName: 'service-a' // name of this application
});
```
## Requirements

> The credentials that your service has requires the following permissions in order to function:

Reviewer 2: ```suggestion
AWS credentials will be loaded as described [here](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials.html). The credentials must have the following permissions:
```

`sqs:DescribeStream` for health checking

`sqs:PutRecord` for placing spans on the queue

## Message encoding

> The message's binary data includes a list of spans. Supported encodings
> are the same as the http [POST /spans](http://zipkin.io/zipkin-api/#/paths/%252Fspans) body.
> 
> Encoding defaults is JSON, now a days we only support this encoding.

Reviewer 2: ```suggestion
The message's binary data includes a list of spans in the same JSON V2 format
as the http [POST /spans](http://zipkin.io/zipkin-api/#/paths/%252Fspans) body.
```

# Related work

[collector-sqs](https://github.com/openzipkin/zipkin-aws/tree/master/collector-sqs)
integrates with zipkin-server to pull spans off of an SQS queue instead

> of http or kafka.

Reviewer 2: ```suggestion
of HTTP or Kafka.
```
